recipes/units/all/conanfile.py

```diff
@@ -0,0 +1,52 @@
+import os
+
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+
+
+class UnitsConan(ConanFile):
+    name = "units"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/nholthaus/units"
+    description = "A compile-time, header-only, dimensional analysis and unit conversion library built on c++14 with no dependencies"
+    topics = ("unit-conversion", "dimensional-analysis", "cpp14",
+              "template-metaprogramming", "compile-time", "header-only",
+              "no-dependencies")
+    license = "MIT"
+    no_copy_source = True
+    settings = "compiler"
+
+    _source_subfolder = "source_subfolder"
+
+    def _compiler_is_supported(self):
+        supported_compilers = [("clang", 3.4), ("gcc", "4.9.3"),
+                               ("Visual Studio", 14.0)]
+        compiler, version = self.settings.compiler, Version(
+            self.settings.compiler.version)
+        return any(compiler == sc[0] and version >= sc[1]
+                   for sc in supported_compilers)
```

Comment:
```suggestion
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, "14")
        minimum_version = {
            "clang": 3.4,
            "gcc": "4.9.3",
            "Visual Studio", 14.0,
        }.get(str(self.settings.compiler))
        if not minimum_version:
             self.output.warn("Unknown compiler: assuminging compiler supports c++14")
        else:
             if Version(self.settings.compiler) < minimum_version:
                 raise ConanInvalidConfiguration("Compiler does not support c++14")
```
I think this code is a bit more readable.
Also, untested.

Reply:
Got it, I put this in.